### DISCUSSION TITLE
Config UI for modules developers (Asking feedback for modules makers)

### DISCUSSION THREAD

**Discussion by Fox2Code** [Fox2Code/FoxMagiskModuleManager#257] [Ideas]
I'm planning to add a config system for those who don't want to worry about making an UI or don't want to add an APK to the system.

I want feedback from developers for what they want, `.xml` files, shell communication, WebView, or apk wrapping.

I'm open to any suggestions.

Please note that I might implement multiple suggestions, as suggestions are not exclusive.

The only requirement is that module developers include a reference to this in their `module.prop`, `config` entry.

**Comment by nift4** [comment]
#69 nice

xml - how do I parse it from a shell script?
apk wrapping - sounds useful for larger modules
webview - very cool idea but again how do i parse from a shell script
shell communication - getting the whole thing done with shell will be pain again

maybe make the xml define the ui and then set a prop?

**Comment by androidacy-user** [comment]
I think a decent approach to support all of these is URI handling for the `config` value
For instance `app://` would indicate an app needs launched for config, `file://` would represent a key=value file to load into the install env, `http://` and `https://` means a website is needed, and perhaps `terminal://` could launch a command in terminal.

**Comment by Fox2Code** [comment]
My app already support `config=packageName` for config button, but I would like to be more open about config screens.

**Comment by silveroxides** [comment]
The option to view and edit how the module is installed or integrate it with other modules similar to how you can edit boot options in grub with a simple command line edit?

**Comment by Fox2Code** [comment]
@silveroxides each module way of choosing how to install itself or integrate itself to other modules is dependent on how the module install script do it, each module do that differently so there is no universal way of doing it.
